python/az/aro/azext_aro/custom.py

```diff
@@ -191,7 +214,62 @@ def aro_list_credentials(client, resource_group_name, resource_name):
     return client.list_credentials(resource_group_name, resource_name)
 
 
-def aro_update(client, resource_group_name, resource_name, no_wait=False):
+def aro_update(cmd, client, resource_group_name, resource_name, no_wait=False):
+    try:
+        oc = client.get(resource_group_name, resource_name)
+
+        master_subnet = oc.master_profile.subnet_id
+        worker_subnets = {w.subnet_id for w in oc.worker_profiles}
+
+        master_parts = parse_resource_id(master_subnet)
+        vnet = resource_id(
+            subscription=master_parts['subscription'],
+            resource_group=master_parts['resource_group'],
+            namespace='Microsoft.Network',
+            type='virtualNetworks',
+            name=master_parts['name'],
+        )
+
+        route_tables = get_route_tables_from_subnets(cmd.cli_ctx, worker_subnets | {master_subnet})
+
+        resources = set()
+        resources.add(vnet)
+        resources.update(route_tables)
+
+        aad = AADManager(cmd.cli_ctx)
+        rp_client_id = os.environ.get('AZURE_FP_CLIENT_ID', FP_CLIENT_ID)
+        rp_client_sp = aad.get_service_principal(rp_client_id)
+
+        c_client_id = oc.service_principal_profile.client_id
+        c_client_sp = aad.get_service_principal(c_client_id)
```

Comment:
if this line throws an exception, try to continue.
if c_client_sp is nil, print a warning (as you are doing).

Reply:
for consistency, please stick to the naming conventions in the file (i.e. client_sp, not c_client_sp).  If you want to change the naming conventions because they're not great (and I agree, they're not great), that's fine but let's do that in a separate PR to avoid further complicating this one.

Reply:
Done - reverted the cluster client ID and SP back to `client_id` and `client_sp`.